src/main/kotlin/lotto/entity/WinLotto.kt

```diff
@@ -1,18 +1,31 @@
 package lotto.entity
 
-class WinLotto(val winNumber: Lotto, val bonus: LottoNumber) {
+import lotto.model.Rank
+
+class WinLotto(private val winNumber: Lotto, private val bonus: LottoNumber) {
     init {
         require(!winNumber.numbers.contains(bonus)) {
-            String.format(
-                ERROR_MESSAGE_DUPLICATED_BONUS_NUMBER,
-                winNumber.toString(),
-                bonus.value
+            ERROR_MESSAGE_DUPLICATED_BONUS_NUMBER.format(
+                winNumber.toString(), bonus.value
             )
         }
     }
 
+    fun makeWinStatistics(purchasedLottos: PurchasedLottos): WinStatistics =
+        WinStatistics(
+            purchasedLottos.value
+                .groupBy { determineRank(it) }
+                .map { it.key to it.value.count() }
+                .toMap()
+        )
```

Comment:
확장 함수를 활용해보려고 노력을 많이 해주셨네요 👍👍
아래 처럼도 작성이 가능하답니다 :)
```suggestion
    return purchasedLottos.value.asSequence()
            .map { determineRank(it) }
            .groupingBy { it }
            .eachCount()
            .let { WinStatistics(it)  }
```

PurchasedLottos에게 당첨 티켓을 넘겨서 결과를 받게 하는 것도 괜찮아 보이네요 :)